### PR TITLE
intel_adsp: ace: cpu clock management

### PR DESCRIPTION
**Pull request by tmleman**
This PR is fixing DSP clock management for ACE platforms. In current state DSP is always using the highest clock.

### CODE REVIEW

**Comment by tmleman** [comment]
I'm adding the DNM label because I haven't tested these changes on lunarlake yet.

**Review by jxstelter — DISMISSED** [review]

File: dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi [hunk 1]
```diff
@@ -85,7 +85,7 @@
 		adsp-clkctl-clk-ipll = <2>;
 		adsp-clkctl-freq-enc = <0xc 0x0 0x4>;
 		adsp-clkctl-freq-mask = <0x0 0x0 0x0>;
-		adsp-clkctl-freq-default = <2>;
+		adsp-clkctl-freq-default = <0>;
```
Comment (kv2019i) [inline dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi:87]: This is interesting, not sure what should be the default policy. We used to run at the highest clock by default previously, why the change now?

File: soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_shim.h [hunk 2]
```diff
@@ -140,7 +140,7 @@ struct ace_dfpmccu {
 
 #endif /* _ASMLANGUAGE */
 
-#define ACE_CLKCTL_WOVCRO    BIT(4)	  /* Request WOVCRO clock */
+#define ACE_CLKCTL_WOVCRO    BIT(21)	  /* Request WOVCRO clock */
```
Comment (kv2019i) [inline soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_shim.h:143]: Can you expand a a bit why this is a safe change to do? Was the code doing incorrect thing before (i.e. this is a fix), or was this in practise unused so this correction has no functional difference?

**Review by kv2019i — COMMENTED** [review]
Looks good for most part. A few clarifiations on the git commits and one open w.r.t. what should be the boot-up frequency be.

File: dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi (hunk 1 above)
Comment (tmleman) [reply dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi:87]: Good question. It turns out that in fact the boot process should take place at the highest clock and when everything is ready, we move to the lowest clock.

File: soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_shim.h (hunk 2 above)
Comment (tmleman) [reply soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_shim.h:143]: This is a fix. don't know where the value 0x10 comes from, but is CLKCTL bit 4 is used for something other than requesting WOVCRO (at least for ACE).

Maybe it's worth asking if anyone knows why we do this exercise with WOVCRO during init?

```c
ACE_DfPMCCU.dfclkctl |= ACE_CLKCTL_WOVCRO;
if (ACE_DfPMCCU.dfclkctl & ACE_CLKCTL_WOVCRO) {
	ACE_DfPMCCU.dfclkctl = ACE_DfPMCCU.dfclkctl & ~ACE_CLKCTL_WOVCRO;
} else {
	platform_lowest_freq_idx = ADSP_CPU_CLOCK_FREQ_LPRO;
}
```

**Review by jxstelter — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]
